src/TestingUtils/Microsoft.AspNetCore.Testing/src/TestFileOutputContext.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    /// <summary>
+    /// Provides access to file storage for the running test. Get access by
+    /// implementing <see cref="ITestMethodLifecycle"/>, and accessing <see cref="TestContext.FileOutput"/>.
+    /// </summary>
+    /// <remarks>
+    /// Requires defining <see cref="AspNetTestFramework"/> as the test framework.
+    /// </remarks>
+    public sealed class TestFileOutputContext
```

Comment:
I tried to centralize as much as possible from `AssemblyTestLog`, but it's not really realistic to make it totally clean. 

This thing is conceptually oriented around test methods and test classes. 
`AssemblyTestLog` lets you pass in arbitrary strings.